src/compas/scene/scene.py

```diff
@@ -47,22 +47,18 @@ def _get_node(self, sceneobject):
         raise Exception("Scene object not in scene")
 
     def clear(self):
-        clear()
+        guids = []
+        for sceneobject in self.sceneobjects:
+            guids += sceneobject.guids
+            sceneobject._guids = None
+        clear(guids=guids)
 
     def redraw(self):
+        self.clear()
+
```

Comment:
@tomvanmele How about now? we collect the all guids of scene then clear all together. Also we add guids as argument for clear, so it does not delete the other stuff in context that are not managed by the scene.